### PR TITLE
refactor(plugins): unify plugin structure

### PR DESCRIPTION
**Pull request by pinglin**
Because

- KrakenD plugin requires absolutely identical Go and package versions, to avoid runtime loading errors, we'd like to keep the plugin dependencies as few as possible

This commit

- unify plugin folder structure
- minimise Go package dependency

### CODE REVIEW

**Review by donch1989 — DISMISSED** [review]

File: plugins/multi-auth/main.go [hunk 1]
```diff
@@ -134,8 +111,36 @@ func (r registerer) registerHandlers(ctx context.Context, extra map[string]inter
...
+	Error(v ...interface{})
+	Critical(v ...interface{})
+	Fatal(v ...interface{})
 }
 
-func main() {}
+// Empty logger implementation
+type noopLogger struct{}
+
+func (n noopLogger) Debug(_ ...interface{})    {}
```
Comment (jvallesm) [inline plugins/multi-auth/main.go:141]: This is repeated 3 times, can we have the interface and an exported `NoopLogger` implementation at `plugins/logger.go`?

**Review by jvallesm — APPROVED** [review]

File: plugins/multi-auth/main.go (hunk 1 above)
Comment (pinglin) [reply plugins/multi-auth/main.go:141]: It's tricky for maintaining KrakenD plugins here. Each folder under `plugins` is a complete plugin package with its own `go.mod`. If we do want to share a functionality across plugins, we can put it in the `x` package instead.

Comment (jvallesm) [reply plugins/multi-auth/main.go:141]: Ah gotcha, I'd seen the modules but hadn't connected that each one would need to import that package. It was a minor non-blocking comment.

In fact, KrakenD points to [this](https://github.com/luraproject/lura/blob/master/logging/log.go#L17) interface, whose package includes a `NoOp` implementation

**Comment by jvallesm** [comment]
Also, I didn't arrive on time for a review in https://github.com/instill-ai/api-gateway/pull/143#pullrequestreview-1945839601 but please do check my comments :)

**Review by jvallesm — CHANGES_REQUESTED** [review]
Could you update the contribution guidelines? the paths there are updated:

```md
$ cd plugins/grpc-proxy && go build -buildmode=plugin -buildvcs=false -o /usr/local/lib/krakend/plugin/grpc-proxy.so /api-gateway/plugins/grpc-proxy && cd /api-gateway # compile the KrakenD grpc-proxy plugin
$ cd plugins/multi-auth && go build -buildmode=plugin -buildvcs=false -o /usr/local/lib/krakend/plugin/multi-auth.so /api-gateway/plugins/multi-auth && cd /api-gateway # compile the KrakenD multi-auth plugin
$ cd plugins/registry && go build -buildmode=plugin -buildvcs=false -o /usr/local/lib/krakend/plugin/registry.so /api-gateway/plugins/registry && cd /api-gateway # compile the registry plugin
```

**Review by jvallesm — DISMISSED** [review]
Unblocking, review comments will be addressed in a [separate PR](https://github.com/instill-ai/api-gateway/pull/148/commits/ba2089889eee22297ad1e64c9ea3a6eacf486967)

**Review by jvallesm — APPROVED** [review]

**Review by jvallesm — APPROVED** [review]

**Pull request merged**